servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/TaskBasedAsyncCompletableOperator.java

```diff
@@ -92,7 +109,11 @@ public final void onSubscribe(final Cancellable cancellable) {
                 LOGGER.trace("executing {} onSubscribe on {}",
                         this instanceof CompletableSubscriberOffloadedTerminals ? "Completable" : "Single",
                         executor);
-                executor.execute(this::deliverSignals);
+                if (offload.getAsBoolean()) {
+                    executor.execute(this::deliverSignals);
+                } else {
+                    deliverSignals();
+                }
             } catch (Throwable t) {
                 // As a policy, we call the target in the calling thread when the executor is inadequately
                 // provisioned. In the future we could make this configurable.
```

Comment:
I missed these in previous reviews, can you remove the debugging log statements below (`LOGGER.trace("offloaded on {}", executor);`, `LOGGER.trace("terminated for {}", executor);`, etc.)?